### ISSUE TITLE
yolov2模型生成问题

### ISSUE THREAD

**Issue by jOHn1016CN** [Dragon-GCS/Python#2]
我的Keras中的BatchNormalization不能够接收weights这个参数，没办法正常构建，以下是traceback截图，求助大佬www
<img width="924" alt="屏幕截图 2024-08-13 012009" src="https://github.com/user-attachments/assets/6e1e58f6-161e-4726-87c1-47e2ea70dcf4">

**Comment by Dragon-GCS** [comment]
图片里的代码信息不全。keras库可能有更新，这是最新的文档https://keras.io/api/layers/normalization_layers/batch_normalization/

**Comment by jOHn1016CN** [comment]
> 图片里的代码信息不全。keras库可能有更新，这是最新的文档https://keras.io/api/layers/normalization_layers/batch_normalization/

谢谢您的回复，keras文档中的BatchNormalization类似乎已经没有了weights这个参数，在您提供的yolov2模型生成的yad2k.py文件中，实例化该类时输入了weights:这一参数导致程序无法正常执行，请问是否有新的解决方法呢

**Comment by Dragon-GCS** [comment]
文件里的代码有点老了，而且不是我写的，有两种办法可以尝试下：
1. 下载老版本的有weights的keras，在[pypi](https://pypi.org/project/keras/#history)上找一下21年左右发布的版本
2. 现在版本有个[`set_weights`方法](https://keras.io/api/layers/base_layer/#setweights-method)，改用这个方法设置权重而不是在初始化层的时候设置。

**Comment by jOHn1016CN** [comment]
好的，谢谢您的回复
